src/garage/torch/algos/pearl.py

```diff
@@ -1,6 +1,635 @@
+        ]
+
+    def get_exploration_policy(self):
+        """Return a policy used before adaptation to a specific task.
+
+        Each time it is retrieved, this policy should only be evaluated in one
+        task.
+
+        Returns:
+            garage.Policy: The policy used to obtain samples that are later
+                used for meta-RL adaptation.
+        """
+        return self.policy
+
+    def adapt_policy(self, exploration_policy, exploration_trajectories):
+        """Produce a policy adapted for a task.
+
+        Args:
+            exploration_policy (garage.Policy): A policy which was returned
+                from get_exploration_policy(), and which generated
+                exploration_trajectories by interacting with an environment.
+                The caller may not use this object after passing it into this
+                method.
+            exploration_trajectories (garage.TrajectoryBatch): Trajectories to
+                adapt to, generated by exploration_policy exploring the
+                environment.
+
+        Returns:
+            garage.Policy: A policy adapted to the task represented by the
+                exploration_trajectories.
+
+        """
+        context = self.sample_context(self._task_idx)
+        self.policy.infer_posterior(context)
+
+        return self.policy
+
+    def to(self, device=None):
+        """Put all the networks within the model on device.
+
+        Args:
+            device (str): ID of GPU or CPU.
+        """
+        if device is None:
+            device = tu.global_device()
+        for net in self.networks:
+            net.to(device)
+
+    @classmethod
+    def augment_env_spec(cls, env_spec, latent_dim):
+        """Augment environment by a size of latent dimension.
+
+        Args:
+            env_spec (garage.envs.EnvSpec): Environment specs to be augmented.
+            latent_dim (int): Latent dimension.
+
+        Returns:
+            garage.envs.EnvSpec: Augmented environment specs.
+
+        """
+        obs_dim = int(np.prod(env_spec.observation_space.shape))
+        action_dim = int(np.prod(env_spec.action_space.shape))
+        aug_obs = akro.Box(low=-1,
+                           high=1,
+                           shape=(obs_dim + latent_dim, ),
+                           dtype=np.float32)
+        aug_act = akro.Box(low=-1,
+                           high=1,
+                           shape=(action_dim, ),
+                           dtype=np.float32)
+        return EnvSpec(aug_obs, aug_act)
+
+    @classmethod
+    def get_env_spec(cls, env_spec, latent_dim, module):
+        """Get environment specs of encoder with latent dimension.
+
+        Args:
+            env_spec (garage.envs.EnvSpec): Environment specs.
+            latent_dim (int): Latent dimension.
+            module (str): Module to get environment specs for.
+
+        Returns:
+            garage.envs.EnvSpec: Encoder environment specs with latent
```

Comment:
looks like this function returns `InOutSpec` (EnvSpec is a subclass)

it might be more clear if you just split these.

also, if this isn't used from the launcher, it can just be private.